### ISSUE TITLE
Installing broken packages fails due to unexpected resolution

### ISSUE THREAD

**Issue by ysndr** [flox/flox#2096]
**Describe the bug:**

Resolving a package that is marked as `broken` in the catalog, will fail on acount of "too tight constraints", as the catalog-service tries to resolve with `allowBroken` as a required constraint.

Previously, the catalog service would return broken packages in a resolution leaving the validation of returned packages to the CLI.

This issue was discovered when looking for a new broken package to replace the now unbroken `tabula` package in #2088.

**Steps to reproduce:**
1. `flox init`
2. `flox install zim`

Result:

```
❌ ERROR: resolution failed: constraints for group 'toplevel' are too tight

   Use 'flox edit' to adjust version constraints in the [install] section,
   or isolate dependencies in a new group with '<pkg>.pkg-group = "newgroup"'
```

Expected:

```
❌ ERROR: Package 'zim' is marked as broken ...
```

**Proposed solution**

To return to the _old_ behavior, the CLI should send requests with `allowBroken = true` by default and verify the response based on the options found in the manifest. Since the verification logic is already in place, only the request generation needs to be changed.

The holistic solution would be to either 
- send a request _with_ constraints, and upon failure, send another unconstrained request and verify the response.
- provide alternative pages server side and validate + pick a page on the cli -- the details on this are less clear.

Both solutions are much more invasive and might be considered after a simpler (interim?) solution has been implemented.

**Flox Version (run `flox --version` if possible):**

1.3.1

**Comment by ysndr** [comment]
at least partially related: https://github.com/flox/flox/issues/1743

**Comment by billlevine** [comment]
> So I think we need to set `allowBroken` to whatever we have in the `allows` block in the manifest, and that has to be checked serverside

This does make sense, and is the current behavior.  I think the primary _problematic_ path is if `allow_broken = false` is set, which is the default, and resolution fails, there is a generic `Constraints too tight message`.  This is not helpful, nor does it allow the CLI any way to identify the reason and instruct the user further.

I think a good solution to this would be to add explicit messaging to the server to report if this is the issue with resolution.  `ResoultionConstraint.Broken` or something that the CLI can use to provide the messaging to the user, suggesting to set `allow_broken = true` to allow resolution to occur.

The other scenario is if `allow_broken = false` forces a very old page which a user _may_ want to be aware of.  This scenario could be helped with the above message, as well as responding with additional pages.  Perhaps the server could be prompted to respond with the latest possible page with the messages _why_ it wasn't selected, along with the latest complete page. This would allow the CLI to present something like `Resolution completed with an old page from <date>.  The newest  page <date> was not used because of ......`.

**Comment by mkenigs** [comment]
Agreed, I think that's a dup of https://github.com/flox/catalog-server/issues/106

**Comment by mkenigs** [comment]
Maybe I should close that issue?

**Comment by ysndr** [comment]
Yes the two issues are likely equivalent 
We don't test the unfree behavior I think but the source issue is the same.

Personally I also don't think that getting a newer package that is broken if there are older version that are not broken is that bad (especially since older instances of the same package version might be broken but not yet tagged as such)

For that to be a bit clearer flox show should probably show that metadata as well

**Comment by ysndr** [comment]
Also should we prefer newer instances of the same version of a package in general? I think that is not taken into account at all yet

**Comment by mkenigs** [comment]
> Also should we prefer newer instances of the same version of a package in general? I think that is not taken into account at all yet

I believe we resolve to the most recent page that satisfies constraints

**Comment by mkenigs** [comment]
> Personally I also don't think that getting a newer package that is broken if there are older version that are not broken is that bad (especially since older instances of the same package version might be broken but not yet tagged as such)

I'm not sure that would ever happen unless you had explicitly set `allowUnfree = true`, in which case wouldn't it be desirable?

**Comment by billlevine** [comment]
> I'm not sure that would ever happen unless you had explicitly set `allowUnfree = true`, in which case wouldn't it be desirable?

Suppose you try to install something, and b/c of the defaults ( `allow_broken = false`) and what you try to install, you end up with a set of packages (b/c everything is in the same `toplevel` group) that is 2.5 years old when you first start using flox. That would be concerning to me and might turn me off if I was not told "Hey, you're getting very old stuff b/c of XYZ. If you want newer, try ABC".  How often this would come up is another question and may very well be _quite unlikely_ , but that's the issue I was trying to convey.
